source/Test.cpp

```diff
@@ -576,8 +580,20 @@ void Test::Fail(const TestContext &context, const PlayerInfo &player, const stri
 	else
 		Logger::LogError("No test conditions were set at the moment of failure.");
 
+	// If this test is supposed to have failed, don't make it fail.
```

Comment:
"it" is a bit ambiguous here; the test still fails.
```suggestion
	// If this test was expected to fail, then return a success exitcode from the program because the test did what it was expected to do.
```